angular/playground/projects/ng-entity-service/src/lib/ng-entity-service.spec.ts

```diff
@@ -0,0 +1,1435 @@
+      [TestServiceWithInlineConfig, HttpTestingController, NgEntityServiceNotifier],
+      (service: TestServiceWithInlineConfig, httpMock: HttpTestingController, notifier: NgEntityServiceNotifier) => {
+        spyOn(notifier, 'dispatch');
+        const entityId = 1;
+        const dummyEntity: Partial<TestEntity> = { foo: 'foo', bar: 123 };
+        service.update(entityId, dummyEntity).subscribe({
+          error: () =>
+            expect(notifier.dispatch).toHaveBeenCalledWith(
+              jasmine.objectContaining({
+                storeName,
+                type: 'error',
+                method: HttpMethod.PUT,
+                errorMsg: '' // !WARN this is inconsistant with the successMsg, as that is undefined
+              })
+            )
+        });
+
+        const req = httpMock.expectOne(`${service.api}/${entityId}`);
+        req.flush([], { status: 500, statusText: '🧨😨💥' });
+      }
+    ));
+
+    it('should dispatch error notice with configured error message when errorMsg is configured and request gives an error', inject(
+      [TestServiceWithInlineConfig, HttpTestingController, NgEntityServiceNotifier],
+      (service: TestServiceWithInlineConfig, httpMock: HttpTestingController, notifier: NgEntityServiceNotifier) => {
+        spyOn(notifier, 'dispatch');
+        const errorMsg = 'Error test message...';
+        const entityId = 1;
+        const dummyEntity: Partial<TestEntity> = { foo: 'foo', bar: 123 };
+        service.update(entityId, dummyEntity, { errorMsg }).subscribe({
+          error: () =>
+            expect(notifier.dispatch).toHaveBeenCalledWith(
+              jasmine.objectContaining({
+                storeName,
+                type: 'error',
+                method: HttpMethod.PUT,
+                errorMsg
+              })
+            )
+        });
+
+        const req = httpMock.expectOne(`${service.api}/${entityId}`);
+        req.flush([], { status: 500, statusText: '🧨😨💥' });
+      }
+    ));
+
+    it('should dispatch stop loading event when request is complete', fakeAsync(
+      inject(
+        [TestServiceWithInlineConfig, HttpTestingController, NgEntityServiceLoader],
+        (service: TestServiceWithInlineConfig, httpMock: HttpTestingController, loader: NgEntityServiceLoader) => {
+          const spy = spyOn(loader, 'dispatch');
+          const entityId = 1;
+          const dummyEntity: Partial<TestEntity> = { foo: 'foo', bar: 123 };
+          service.update(entityId, dummyEntity).subscribe();
+
+          const req = httpMock.expectOne(`${service.api}/${entityId}`);
+          req.flush([]);
+          tick(); // needed for the finalize pipe operator to run
+          expect(spy.calls.mostRecent().args).toEqual([
+            {
+              storeName,
+              loading: false,
+              entityId, // !WARN this is inconsistant with the stop loading event in both get & add, as they don't include entityId
```

Comment:
Do you think we should add it also to get and add methods?

Reply:
I think that'll be a good idea. Not sure what it's used for, but it can't hurt to be consistent

Reply:
Cool, so you can add it, please.

Reply:
done